### PR TITLE
Get IDataView to C# Path Sorted

### PR DESCRIPTION
**Pull request by wschin**
Fixes partially #1708 (first and second working items).

### CODE REVIEW

File: test/Microsoft.ML.Tests/Scenarios/Api/CookbookSamples/CookbookSamples.cs [hunk 1]
```diff
@@ -66,9 +66,9 @@ private void IntermediateData(string dataPath)
             var transformedData = dataPipeline.Fit(data).Transform(data);
 
             // 'transformedData' is a 'promise' of data. Let's actually read it.
-            var someRows = transformedData.AsDynamic
+            var someRows = mlContext
                 // Convert to an enumerable of user-defined type. 
-                .AsEnumerable<InspectedRow>(mlContext, reuseRowObject: false)
+                .CreateEnumerable<InspectedRow>(transformedData.AsDynamic, reuseRowObject: false)
```
Comment (Ivanidzo4ka) [inline test/Microsoft.ML.Tests/Scenarios/Api/CookbookSamples/CookbookSamples.cs]: >CreateEnumerable [](start = 17, length = 16)

You need to update [https://github.com/dotnet/machinelearning/blob/master/docs/code/MlNetCookBook.md](https://github.com/dotnet/machinelearning/blob/master/docs/code/MlNetCookBook.md) with changes in this filea and other CookBookSamples file.

Comment (wschin) [reply test/Microsoft.ML.Tests/Scenarios/Api/CookbookSamples/CookbookSamples.cs]: Only two examples touched and they are fixed. Please search for `InspectedRow` for static one and `InspectedRowWithAllFeatures` dynamic one.

File: test/Microsoft.ML.Tests/Transformers/WordTokenizeTests.cs [hunk 2]
```diff
@@ -66,7 +67,7 @@ public void WordTokenizeWorkout()
             var result = pipe.Fit(dataView).Transform(dataView);
 
             // Extract the transformed result of the first row (the only row we have because data contains only one TestClass) as a native class.
-            var nativeResult = new List<NativeResult>(result.AsEnumerable<NativeResult>(Env, false))[0];
+            var nativeResult = ML.CreateEnumerable<NativeResult>(result, false).ToList()[0];
```
[the remark this answers is not in the document]
Comment (wschin) [reply test/Microsoft.ML.Tests/Transformers/WordTokenizeTests.cs]: Fixed. Thanks.

File: src/Microsoft.ML.Data/DataView/TypedCursor.cs [hunk 3]
```diff
@@ -544,41 +544,41 @@ public static class CursoringUtils
...
 
-            return TypedCursorable<TRow>.Create(env, data, ignoreMissingColumns, schemaDefinition);
+            return TypedCursorable<TRow>.Create(catalog.Environment, data, ignoreMissingColumns, schemaDefinition);
         }
 
         /// <summary>
         /// Convert an <see cref="IDataView"/> into a strongly-typed <see cref="IEnumerable{TRow}"/>.
         /// </summary>
         /// <typeparam name="TRow">The user-defined row type.</typeparam>
+        /// <param name="mlContext">Context where </param>
```
Comment (Ivanidzo4ka) [inline src/Microsoft.ML.Data/DataView/TypedCursor.cs]: >Context where </p [](start = 36, length = 17)

How did I approve you?

**Pull request merged**